cloudinit/distros/networking.py

```diff
@@ -185,18 +186,47 @@ def try_set_link_up(self, devname: DeviceName) -> bool:
 class BSDNetworking(Networking):
     """Implementation of networking functionality shared across BSDs."""
 
+    def __init__(self):
+        self.ifc = ifconfig.Ifconfig()
+        self.ifs = {}
+
+    def _update_ifs(self):
```

Comment:
I've head this initially, and made lots of tests fail 😅 
so, yes, i think it would make sense… it'll just be more work.
but, yeah, we probably do not need to call `ifconfig -a` seventeen thousand times.
so, yes, a good idea.